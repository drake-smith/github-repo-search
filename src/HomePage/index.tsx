import React from 'react';
import styled from 'styled-components';
import { media } from '../shared/media';
import SearchBar from '../SearchBar';
import SearchResults from '../SearchResults';
import SearchFilters from '../SearchFilters';

interface HomePageProps {
  searchTerm: string;
  handleSearchTermChange: (searchTerm: string) => void;
  handleSearchSubmit: () => void;
  sortBy: string;
  handleSortByChange: (sortBy: string) => void;
  language: string;
  handleLanguageChange: (language: string) => void;
  isLoading: boolean;
  hasError: boolean;
  searchResults: any[];
  hasNoResults: boolean;
  handleSelectSearchResult: (index: number) => void;
}

const ContentContainer = styled.div`
  max-width: 1280px;
  margin: 0 auto;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
`;

const HomePage = ({
  searchTerm,
  handleSearchTermChange,
  handleSearchSubmit,
  sortBy,
  handleSortByChange,
  language,
  handleLanguageChange,
  isLoading,
  hasError,
  searchResults,
  hasNoResults,
  handleSelectSearchResult,
}: HomePageProps) => {
  return (
    <ContentContainer>
      <SearchBar
        searchTerm={searchTerm}
        handleSearchSubmit={handleSearchSubmit}
        handleSearchTermChange={handleSearchTermChange}
        isDisabled={isLoading}
      />
      {hasError && (
        <p>There was an error fetching the search results. Please try again.</p>
      )}
      {hasNoResults && <p>The search returned no results. Please try again.</p>}
      {searchResults.length > 0 && (
        <SearchFilters
          language={language}
          handleLanguageChange={handleLanguageChange}
          sortBy={sortBy}
          handleSortByChange={handleSortByChange}
        />
      )}
      {isLoading ? (
        <p>Fetching results...</p>
      ) : (
        <SearchResults
          results={searchResults}
          handleSelectSearchResult={handleSelectSearchResult}
        />
      )}
    </ContentContainer>
  );
};

export default HomePage;
