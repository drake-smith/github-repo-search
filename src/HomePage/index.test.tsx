// @ts-ignore
import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import HomePage from '.';


describe('<HomePage />', () => {
  const handleSearchTermChange = jest.fn();
  const handleSearchSubmit = jest.fn();
  const handleSortByChange = jest.fn();
  const handleLanguageChange = jest.fn();
  const handleSelectSearchResult = jest.fn();

  test('It renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          {/* @ts-ignore */}
          <HomePage
            searchTerm={''}
            handleSearchTermChange={handleSearchTermChange}
            handleSearchSubmit={handleSearchSubmit}
            sortBy={''}
            handleSortByChange={handleSortByChange}
            language={'javascript'}
            handleLanguageChange={handleLanguageChange}
            isLoading={false}
            hasError={false}
            searchResults={[]}
            hasNoResults={false}
            handleSelectSearchResult={handleSelectSearchResult}
          />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('isLoading UI renders', () => {
    render(
      <MemoryRouter>
        {/* @ts-ignore */}
        <HomePage
          searchTerm={''}
          handleSearchTermChange={handleSearchTermChange}
          handleSearchSubmit={handleSearchSubmit}
          sortBy={''}
          handleSortByChange={handleSortByChange}
          language={'javascript'}
          handleLanguageChange={handleLanguageChange}
          isLoading={true}
          hasError={false}
          searchResults={[]}
          hasNoResults={false}
          handleSelectSearchResult={handleSelectSearchResult}
        />
      </MemoryRouter>
    );

    expect(screen.findByText('Fetching results...')).toBeTruthy();
  });
});
