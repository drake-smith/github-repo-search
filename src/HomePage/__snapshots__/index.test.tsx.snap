// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> It renders correctly 1`] = `
<div
  className="sc-bqiRlB dtxFRl"
>
  <form
    className="sc-bdvvtL jcpPon"
    onSubmit={[Function]}
  >
    <input
      aria-label="Search for GitHub repositories"
      className="sc-gsDKAQ ibryMM"
      onChange={[Function]}
      placeholder="Search for GitHub repositories"
      type="search"
      value=""
    />
    <input
      className="sc-dkPtRN PqHbV"
      disabled={true}
      type="submit"
      value="Search"
    />
  </form>
  <ul
    className="sc-furwcr RmOMa"
  />
</div>
`;
